File: src/main/java/com/OddJob/Location.java
package com.OddJob;


import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

@Entity
public class Location {


    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String street;
    private String zipCode;
    private String city;
    private String country;
    private double lat;
    private double lng;

    public Location(){}

    public Location(String street, String zipCode, String city, String country, double lat, double lng) {
        this.street = street;
        this.zipCode = zipCode;
        this.city = city;
        this.country = country;
        this.lat = lat;
        this.lng = lng;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getStreet() {
        return street;
    }

    public void setStreet(String street) {
        this.street = street;
    }

    public String getZipCode() {
        return zipCode;
    }

    public void setZipCode(String zipCode) {
        this.zipCode = zipCode;
    }

    public String getCity() {
        return city;
    }

    public void setCity(String city) {
        this.city = city;
    }

    public String getCountry() {
        return country;
    }

    public void setCountry(String country) {
        this.country = country;
    }

    public double getLat() {
        return lat;
    }

    public void setLat(double lat) {
        this.lat = lat;
    }

    public double getLng() {
        return lng;
    }

    public void setLng(double lng) {
        this.lng = lng;
    }
}

File: webapp/src/app/models/location.ts
export class Location {
  id: number;
  street: string;
  zipCode: string;
  city: string;
  country: string;
  lat: number;
  lng: number;

  constructor(
    id: number,
    street: string,
    zipCode: string,
    city: string,
    country: string,
    lat: number,
    lng: number,
  ) {
    this.id = id;
    this.street = street;
    this.zipCode = zipCode;
    this.city = city;
    this.country = country;
    this.lat = lat;
    this.lng = lng;
   }
}
